plugins/woocommerce/includes/data-stores/class-wc-coupon-data-store-cpt.php

```diff
@@ -532,16 +532,16 @@ public function check_and_hold_coupon( $coupon ) {
 		);
 
 		$query_for_tentative_usages = $this->get_tentative_usage_query( $coupon->get_id() );
-		$db_timestamp               = $wpdb->get_var( 'SELECT UNIX_TIMESTAMP() FROM DUAL' );
+		$db_timestamp               = $wpdb->get_var( 'SELECT UNIX_TIMESTAMP() FROM ' . $wpdb->posts . ' LIMIT 1' );
 
 		$coupon_usage_key = '_coupon_held_' . ( (int) $db_timestamp + $held_time ) . '_' . wp_generate_password( 6, false );
 
 		$insert_statement = $wpdb->prepare(
 			"
 			INSERT INTO $wpdb->postmeta ( post_id, meta_key, meta_value )
-			SELECT %d, %s, %s FROM DUAL
+			SELECT %d, %s, %s FROM $wpdb->posts
```

Comment:
This could be problematic if the posts table is empty. Of course, I realize this same point was noted in the PR description, and in the linked dicsussion, but I wanted to pause and think about alternative paths.

For instance, based on some light testing against MySQL 5.5 (I couldn't immediately find a way of testing with earlier versions—nothing below 5.5 is available on Docker Hub, for instance), this approach seems to work:

```sql
INSERT INTO test ( a, b, c )
SELECT      'a', 'b', 'c' 
FROM        ( SELECT null ) pseudo_dual
WHERE       (10 + 5 ) > 10 ;
```

This allows for a `WHERE` clause without either `dual` or an existing table. Mapping it back to this line (and, if we adopt this, I suspect we could drop the `LIMIT` clause), we get:

```suggestion
			SELECT %d, %s, %s FROM ( SELECT null ) pseudo_dual
```

What do we think? Or, could this be problematic in different ways?

Reply:
Thank you for testing and for looking into this! I'm afraid I had left out an important detail when creating this PR. I touched on it above just now, but one of the other requirements is that each query uses a table which the host recognizes. You're right that this approach would work without using `DUAL`, but something I left out was that it needs to still use a table which the host can recognize. Sorry for the confusion.

> ..based on some light testing against MySQL 5.5.. 

When first discussing this with @vedanshujain, he said that he thought he remembered that DUAL was added to prevent issues caused by not specifying DUAL in versions of MySQL below 5.5. If that's the case, we might need to specify DUAL or an existing table to avoid those issues.

> This allows for a WHERE clause without either dual or an existing table.

This is an interesting finding. I tried using this technique in place of `SELECT UNIX_TIMESTAMP()` (resulting in something `SELECT UNIX_TIMESTAMP() FROM ( SELECT NULL ) pseudo_dual`), but this also failed in the same way on this host.

Reply:
Ah, ok, thanks for confirming (so, behind the scenes, are we talking about HyperDB here?) ... it would seem  we don't have too many options, then, but let me think a little about whether this is something we're comfortable with applying to *all* installations.

Reply:
Correct, I am referring to HyperDB, here. See also: pMz3w-hP4-p2.

Reply:
Note that `SELECT %d, %s, %s FROM ( SELECT null ) pseudo_dual` is same as `SELECT %d, %s, %s FROM ( SELECT null FROM DUAL) pseudo_dual` so we are technically still specifying DUAL (or more accurately, in both cases we are not specifying DUAL). Dual is just a convenience keyword where no table is specified, so in recent MySQL versions `DUAL` is anyway removed by the interpreter.